internal/api/clairintegration/api.go

```diff
@@ -71,3 +95,95 @@ func (a *API) reverseProxyToClair(w http.ResponseWriter, r *http.Request) {
 
 	respondwith.JSON(w, http.StatusOK, responseBody)
 }
+
+func (a *API) handleClairNotification(w http.ResponseWriter, r *http.Request) {
+	httpapi.IdentifyEndpoint(r, "/clair-notification")
+
+	c := a.cfg.ClairClient
+
+	// retrieve and parse notification
+	hasSecretHeader := false
+auth:
+	for header, value := range r.Header {
+		// go camelCases headers
+		if strings.ToUpper(header) == "X-KEPPEL-CLAIR_NOTIFICATION-SECRET" {
+			for _, entry := range value {
+				if entry == c.NotificationSecret {
+					hasSecretHeader = true
+					break auth
+				}
+			}
+		}
+	}
+
+	if !hasSecretHeader {
+		http.Error(w, "unauthorized", http.StatusUnauthorized)
+		return
+	}
+
+	reqBodyBytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnprocessableEntity)
+		logg.Error(err.Error())
+		return
+	}
```

Comment:
IO errors should be a 5xx in my opinion.
```suggestion
if respondwith.ErrorText(w, err) {
  return
}
```

Reply:
This was marked as resolved without any change.